opendbc/car/tesla/values.py

```diff
@@ -34,6 +34,36 @@ class TeslaCarDocsHW4(CarDocs):
 class TeslaPlatformConfig(PlatformConfig):
   dbc_dict: DbcDict = field(default_factory=lambda: {Bus.party: 'tesla_model3_party'})
 
+def match_fw_to_car_fuzzy(live_fw_versions: LiveFwVersions, vin: str, offline_fw_versions: OfflineFwVersions) -> set[str]:
+  candidates: set[str] = set()
+  for candidate, fws in offline_fw_versions.items():
+    for ecu, expected_versions in fws.items():
+      addr = ecu[1:]
+
+      current_model = decode_tesla_fingerprint_model(expected_versions[0])
+      if current_model is None:
+        continue
+
+      live_current_fw = live_fw_versions.get(addr, set())
+      if not live_current_fw:
+        continue
+      live_current_fw = list(live_current_fw)[0]
+
+      live_model = decode_tesla_fingerprint_model(live_current_fw)
+      if live_model is None:
+        continue
+      if live_model == current_model:
+        candidates.add(candidate)
+  return candidates
+
+def decode_tesla_fingerprint_model(fingerprint: bytes) -> tuple:
+  if not fingerprint.startswith(b'TeM'): # Not a tesla
+    return None
+  model_split = fingerprint.split(b'),')
+  if len(model_split) != 2:
+    return None
+  model = model_split[1][0]
```

Comment:
this should be robust to garbage input, i.e. no Python exceptions. We use a simple regex pattern for other brands, it ensures the parsing logic is in one place and if there's no match, it's obvious what it will return (None)